includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -1474,19 +1560,24 @@ class="jetpack-logo"
 	public function wc_setup_activate_save() {
 		check_admin_referer( 'wc-setup' );
 
-		// Determine if we need to install Jetpack synchronously here.
+		// Clean up temporary Jetpack queued install option.
+		// This happens after the connection button is clicked
+		// and we waited for the pending install to finish.
+		delete_option( 'woocommerce_setup_queued_jetpack_install' );
```

Comment:
The JavaScript loaded in the wizard (https://github.com/woocommerce/woocommerce/blob/master/assets/js/admin/wc-setup.js#L96) is supposed to prevent the activate step's form from submitting if a pending install is taking place.

Why does the option need to be cleared here?

Reply:
Once the form is submitted (or if it wasn't blocked initially), `wc_setup_activate_save` is executed. At this point, we no longer care whether the Jetpack install is queue, so we should clear the option.

Reply:
> or if it wasn't blocked initially

Could that happen in a normal flow, or did it require mucking with database options and the like?

Reply:
@jeffstieler yes, it can happen initially. The form is blocked when both of these conditions are met:
1. If the Jetpack class is not active:
The Jetpack class is not active under two scenarios: Jetpack is installed but not active, or Jetpack is not installed.
2. if the `woocommerce_setup_queued_jetpack_install` was set earlier.

The option `woocommerce_setup_queued_jetpack_install` is not saved if the user didn't select any WCS-related services.

Regardless, I still think that the option should be removed in `wc_setup_activate_save` even if the form is blocked each time. There's no reason to keep around a temporary option like that.

Are you thinking something different?